src/scene/gsplat/gsplat-instance.js

```diff
@@ -212,10 +222,30 @@ class GSplatInstance {
             const camera = this.cameras[0];
             this.sort(camera._node);
 
+            // resolve spherical harmonics
+            this.resolveSH?.render(camera._node, this.meshInstance.node.getWorldTransform());
+
             // we get new list of cameras each frame
             this.cameras.length = 0;
         }
     }
+
+    setFullSH(value) {
+        const { resource } = this;
+        const { gsplatData } = resource;
+
+        if (gsplatData instanceof GSplatSogsData && gsplatData.shBands > 0) {
+
+            if (this.resolveSH) {
```

Comment:
instead of handling if the value is different in the component, I'd maybe always call this function in the instance, and handle that here. Otherwise it's strange to see that we delete resolveSH regardless of the value, as that assumes the component does not call this function unless the change is needed.